src/libraries/System.Text.RegularExpressions/tests/RegexSRMTests.cs

```diff
@@ -37,6 +37,43 @@ public void TestBoundary()
             Assert.True(Regex.IsMatch(" AB\u200dCD ", @"\b\w+\b"));
         }
 
+        public static IEnumerable<object[]> TestLazyLoops_Data()
```

Comment:
presumably we want to end up with only SRM-specific tests (which are hopefully relatively few) in the RegexSRMTests.cs file. 

could clean up in a followup if you prefer.

Reply:
Right, I've been making some tests apply to None and Compiled as well each time, to get more confidence in expected result. Thus, they've become not just DFA tests. In the end those test could indeed be moved to where they best belong -- I would say -- as a follow-up cleanup. In some cases I'm not sure where they best belong.

Reply:
looks like most or all of RegexSRMTests.cs is relevant to the other engine. And more importantly than where it lives, it would be good to make those tests run against the existing engine as another test against discrepancies.

Reply:
I'll go over the tests after the bug fix and generalize them. There are some very specific DFA test though.